packages/ra-ui-materialui/src/list/filter/FilterListItem.tsx

```diff
@@ -148,23 +149,21 @@ const FilterListItem: FC<{ label: string; value: any }> = props => {
     const translate = useTranslate();
     const classes = useStyles(props);
 
-    const isSelected = isEqual(value, filterValues);
+    const isSelected =
+        isEmpty(value) || isEmpty(filterValues)
+            ? false
+            : isMatch(filterValues, value);
 
     const addFilter = () => {
-        setFilters({ ...filterValues, ...value }, null, false);
```

Comment:
Problem : we should update URL properly with the nested fields of value and without overriding nodes of other filters contained in filterValues. 
Correction : using lodash deep merge instead of spread operator.